src/app/lib/hooks/useIsAdvancedUser.ts

```diff
@@ -66,10 +67,10 @@ const useIsAdvancedUser = () => {
      * PROD: only allowlist
      * TEST: more than 1 token or allowlist
      */
-    onSuccess: (balance) => {
+    onSuccess: (balance: bigint) => {
```

Comment:
cc @stepandel @kent 

Once I replaced contract-specific classes (e.g., `OptimismToken`) with generic interfaces (e.g., `IContractToken`), I realized that return types were not being automatically inferred, hence the type casting like `balance: bigint`. If you guys know a way to infer types from interfaces, please let me know ;)

Reply:
the `onSuccess()` type is inferred from the `abi` type. Since it's `any`, it won't be able detect the types for either `args` or `onSuccess()`

You can technically write an `AbiType` for each `ContractType` if it's worth the effort.

One approach could be have a generic ABI for Token & governor contracts, maybe even alligator, which is a stripped down ABI with only a few methods that are actually in use: https://github.com/voteagora/agora-next/blob/aebddb3332b02ebeb5920510c72bcb66176decfe/src/lib/tenant/tenantContractFactory.ts#L42

Reply:
@stepandel I see. I think building a generic abi makes sense. I can add it to my todo list for the `ens` implementation.